### PR TITLE
Remove signup iframe

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
The signup form is currently displayed within an iframe. Removing the iframe helps to reduce the complexity of the DCR codebase  and the decision behind this change can be found [here](https://docs.google.com/document/d/1nKKDxOv0U94EmZ7rqBJdc9ffYFLX5i_gy3tD6BuZiio/edit). 

## Why?
Reduce the complexity of the codebase. 

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |

[before]: https://github.com/guardian/dotcom-rendering/assets/20416599/3ec9a20d-a140-4994-afa0-a82c18b16566
[after]: https://github.com/guardian/dotcom-rendering/assets/20416599/7f06f878-897d-4475-af2f-d86c8d9ed2e0

### CODE REVIEW

File: dotcom-rendering/src/layouts/NewsletterSignupLayout.tsx [hunk 1]
```diff
@@ -445,20 +445,19 @@ export const NewsletterSignupLayout = ({ article, NAV, format }: Props) => {
...
-										}
-										hidePrivacyMessage={true}
-									/>
-
-									<Hide from="desktop">
-										<NewsletterPrivacyMessage />
-									</Hide>
+									<Island
+										priority="feature"
+										defer={{ until: 'idle' }}
```
Comment (mxdvl) [inline dotcom-rendering/src/layouts/NewsletterSignupLayout.tsx:450]:
```suggestion
										defer={{ until: 'visible' }}
```

**Review by mxdvl — COMMENTED** [review]
It seems like the Island is SecureSignup, but we've made its parent WebSignup importable instead. What's the correct intent?

**Comment by abeddow91** [comment]
TLDR: Ive refactored this to make the signup component an island. 

Its a good point @mxdvl and I've refactored to help this. It had previously been the parent so that we could check the bridget compatiblity in the parent of the apps wrapper (and iirc you cant nest islands in islands). However, 1) we dont need to check for this bridget version any more as we have enough users on this version and 2) we don't actually support newsletters on apps yet.

File: dotcom-rendering/src/components/SecureSignup.importable.tsx [hunk 2]
```diff
@@ -311,7 +308,15 @@ export const SecureReCAPTCHASignup = ({
...
 		<>
-			<form onSubmit={handleSubmit} id={`secure-signup-${newsletterId}`}>
+			<form
+				onSubmit={handleSubmit}
+				id={`secure-signup-${newsletterId}`}
+				css={css`
+					display: ${hasResponse || isWaitingForResponse
+						? 'none'
+						: 'block'};
+				`}
```
Comment (mxdvl) [inline dotcom-rendering/src/components/SecureSignup.importable.tsx:318]: This can be a direct `style` prop as per Emotion best practice.

File: dotcom-rendering/src/layouts/NewsletterSignupLayout.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/layouts/NewsletterSignupLayout.tsx:450]: Un resolving as it's `visible` for the other Island. Why the distinction.

Comment (abeddow91) [reply dotcom-rendering/src/layouts/NewsletterSignupLayout.tsx]: Ah, there shouldn't be a distinction - they should both be visible. I've updated this

File: dotcom-rendering/src/components/SecureSignup.importable.tsx [hunk 3]
```diff
@@ -212,20 +235,20 @@ const sendTracking = (
...
-export const SecureReCAPTCHASignup = ({
-	newsletterId,
-	successDescription,
-}: Props) => {
+export const SecureSignup = ({ newsletterId, successDescription }: Props) => {
 	const recaptchaRef = useRef<ReactGoogleRecaptcha>(null);
 	const [captchaSiteKey, setCaptchaSiteKey] = useState<string>();
+	const [signedInUserEmail, setSignedInUserEmail] = useState<
+		string | undefined
+	>(undefined);
```
Comment (mxdvl) [inline dotcom-rendering/src/components/SecureSignup.importable.tsx:251]: FYI – types can be inferred

```suggestion
	const [signedInUserEmail, setSignedInUserEmail] = useState<string>();
```

File: dotcom-rendering/src/components/SecureSignup.importable.tsx [hunk 4]
```diff
@@ -237,6 +260,9 @@ export const SecureReCAPTCHASignup = ({
 
 	useEffect(() => {
 		setCaptchaSiteKey(window.guardian.config.page.googleRecaptchaSiteKey);
+		void resolveEmailIfSignedIn().then((email) => {
+			setSignedInUserEmail(email);
+		});
```
Comment (mxdvl) [inline dotcom-rendering/src/components/SecureSignup.importable.tsx:265]: Not a blocker, but maybe simpler?

```suggestion
		void resolveEmailIfSignedIn().then(setSignedInUserEmail);
```

File: dotcom-rendering/src/components/SecureSignup.importable.tsx (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/SecureSignup.importable.tsx:346]: This is still true…

File: dotcom-rendering/src/components/SecureSignup.importable.tsx [hunk 5]
```diff
@@ -308,23 +334,46 @@ export const SecureReCAPTCHASignup = ({
...
+					`}
 				/>
 
 				<div css={flexParentStyles}>
-					<div css={inputContainerStyles}>
+					<div
+						css={css`
+							${inputContainerStyles};
+							flex-basis: ${!signedInUserEmail ? '335px' : '0'};
+						`}
```
Comment (mxdvl) [inline dotcom-rendering/src/components/SecureSignup.importable.tsx:363]: I think here we’d want an array of `SerialisedStyles`?

```
css={[inputContainerStyles, css`flex-basis: /* … */;`]}
```

File: dotcom-rendering/src/lib/contributions.ts [hunk 6]
```diff
@@ -249,6 +251,37 @@ export const recentlyClosedBanner = (
...
+			setTimeout(() => resolve(null), 1000);
+			void getEmail(idapiUrl).then((email) => {
+				if (email) {
+					resolve(email);
+				} else {
+					resolve(null);
+				}
+			});
+		});
+	};
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/contributions.ts:283]: I think this is already defined in `fetchEmail.ts`

**Review by mxdvl — COMMENTED** [review]
Nearly there!

File: dotcom-rendering/src/components/SecureSignup.importable.tsx [hunk 7]
```diff
@@ -308,23 +330,52 @@ export const SecureReCAPTCHASignup = ({
...
+			>
 				<Label
 					text="Enter your email address"
-					cssOverrides={labelStyles}
+					cssOverrides={css`
+						${labelStyles};
+						display: ${!signedInUserEmail
+							? 'inline-block'
+							: 'none'};
+					`}
```
Comment (mxdvl) [inline dotcom-rendering/src/components/SecureSignup.importable.tsx:351]: Not a blocker, but we could use an array of `SerialisedStyles` here

File: dotcom-rendering/src/layouts/NewsletterSignupLayout.tsx [hunk 8]
```diff
@@ -445,20 +445,22 @@ export const NewsletterSignupLayout = ({ article, NAV, format }: Props) => {
...
+											<SecureSignup
+												newsletterId={
+													promotedNewsletter.identityName
+												}
+												successDescription={
+													promotedNewsletter.successDescription
+												}
+											/>
+											<NewsletterPrivacyMessage />
+										</>
```
Comment (mxdvl) [inline dotcom-rendering/src/layouts/NewsletterSignupLayout.tsx:462]: I do not think you can have a child of island be anything but a single component that has `*.importable.tsx` in its filename
